crates/primitives/src/env.rs

```diff
@@ -1,50 +1,47 @@
-pub mod handler_cfg;
-
-use alloy_primitives::TxKind;
-pub use handler_cfg::{CfgEnvWithHandlerCfg, EnvWithHandlerCfg, HandlerCfg};
+use derive_where::derive_where;
 
 use crate::{
-    calc_blob_gasprice, AccessListItem, Account, Address, Bytes, InvalidHeader, InvalidTransaction,
-    Spec, SpecId, B256, GAS_PER_BLOB, KECCAK_EMPTY, MAX_BLOB_NUMBER_PER_BLOCK, MAX_INITCODE_SIZE,
-    U256, VERSIONED_HASH_VERSION_KZG,
+    block, calc_blob_gasprice, AccessListItem, Account, Address, Block, Bytes, ChainSpec,
+    InvalidHeader, InvalidTransaction, Spec, SpecId, TransactionValidation, B256, KECCAK_EMPTY,
+    MAX_BLOB_NUMBER_PER_BLOCK, MAX_INITCODE_SIZE, U256, VERSIONED_HASH_VERSION_KZG,
+};
+use crate::{
+    transaction::{self, Transaction},
+    TxKind,
 };
 use core::cmp::{min, Ordering};
+use core::fmt::Debug;
 use core::hash::Hash;
 use std::boxed::Box;
 use std::vec::Vec;
 
 /// EVM environment configuration.
-#[derive(Clone, Debug, Default, PartialEq, Eq)]
+#[derive_where(Clone, Debug, Default; ChainSpecT::Block, ChainSpecT::Transaction)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-pub struct Env {
+pub struct Env<ChainSpecT: ChainSpec> {
     /// Configuration of the EVM itself.
     pub cfg: CfgEnv,
     /// Configuration of the block the transaction is in.
-    pub block: BlockEnv,
+    pub block: ChainSpecT::Block,
     /// Configuration of the transaction that is being executed.
-    pub tx: TxEnv,
```

Comment:
As `BlockEnv` and `TxEnv` are no needed for the `EthChainSpec`, should they be moved to `chain_spec.rs`?